src/avl.ml

```diff
@@ -441,6 +441,8 @@ let avl_push
       (mem, LeafPtr leaf_ptr)
   end
 
+type queue_end = Front | Back
```

Comment:
Is there a better place for this type given that it is not used anywhere in this module?

Reply:
Ya at first I thought this might be useful in avl.ml, but since this makes sense. Moved it over to sliceList.ml for the time being.